examples/motor-imagery/plot_ACM.py

```diff
@@ -0,0 +1,282 @@
+    'augmenteddataset__lag': [1, 2, 3, 4, 5, 6, 7],
+}
+
+param_grid["ACM(Grid)+MDM"] = {
+    'augmenteddataset__order': [1, 2, 3, 4, 5, 6, 7],
+    'augmenteddataset__lag': [1, 2, 3, 4, 5, 6, 7],
+}
+
+pipelines_grid = {}
+pipelines_grid["ACM(Grid)+TGSP+SVM"] = GridSearchCV(
+    pipelines["ACM(Grid)+TGSP+SVM"],
+    param_grid["ACM(Grid)+TGSP+SVM"],
+    refit=True,
+    cv=inner_cv,
+    n_jobs=-1,
+    scoring="accuracy",
+)
+
+pipelines_grid["ACM(Grid)+MDM"] = GridSearchCV(
+    pipelines["ACM(Grid)+MDM"],
+    param_grid["ACM(Grid)+MDM"],
+    refit=True,
+    cv=inner_cv,
+    n_jobs=-1,
+    scoring="accuracy",
+)
+
+# Run the inner CV for getting the hyper-parameter
+results_grid = []
+best_estimator = []
+for ppn, ppl in pipelines_grid.items():
+    cvclf = clone(ppl)
+    score = cvclf.fit(X_train, y_train)
+    res = {
+        "pipeline": ppn,
+        "order": score.best_params_["augmenteddataset__order"],
+        "lag": score.best_params_["augmenteddataset__lag"]
+    }
+
+    res_best = {
+        "pipeline": ppn,
+        "best_estimator": score.best_estimator_
+    }
+    best_estimator.append(res_best)
+    results_grid.append(res)
+
+results_grid = pd.DataFrame(results_grid)
+best_estimator = pd.DataFrame(best_estimator)
+print(results_grid)
+
+# Update the pipeline with the best pipeline obtained with GridSearch process
+pipelines["ACM(Grid)+TGSP+SVM"] = best_estimator.loc[
+    best_estimator['pipeline'] == "ACM(Grid)+TGSP+SVM",
+    "best_estimator"].values[0]
+
+pipelines["ACM(Grid)+MDM"] = best_estimator.loc[
+    best_estimator['pipeline'] == "ACM(Grid)+MDM",
+    "best_estimator"].values[0]
+
+###############################################################################
+# Defining pipelines
+# ------------------
+#
+# Compare TGSP+SVM, MDM versus ACM(Grid)+TGSP+SVM and ACM(Grid)+MDM
+
+# Define the standard pipeline TGSP+SVM as baseline with standard covariance
+pipelines["TGSP+SVM"] = Pipeline(steps=[
+    ("Covariances", Covariances("oas")),
+    ("Tangent_Space", TangentSpace(metric="riemann")),
+    ("SVM", SVC(kernel="linear"))
+])
+
+pipelines["MDM"] = Pipeline(steps=[
+    ("Covariances", Covariances("cov")),
+    ("MDM", MDM(metric=dict(mean='riemann', distance='riemann')))
+])
+
+###############################################################################
+# Evaluation
+# ----------
+
+results = []
+for ppn, ppl in pipelines.items():
+    cvclf = clone(ppl)
+    cvclf.fit(X_train, y_train)
+    y_pred = cvclf.predict(X_test)
+    score = accuracy_score(y_test, y_pred)
+    res = {
+        "score": score,
+        "pipeline": ppn,
+    }
+    results.append(res)
+
+results = pd.DataFrame(results)
+
+# Calculate the mean score for each pipelines
+pipeline_stats = results.groupby('pipeline')['score'].agg(
+    ['mean']).sort_values(by='mean', ascending=False)
+
+for pipeline, stats in pipeline_stats.iterrows():
+    print(f"{pipeline}, score: {stats['mean']:.4f}")
```

Comment:
There is no longer any need to calculate the mean. Code can be simplified into:
```suggestion
for _, row in results.sort_values(by='score', ascending=False).iterrows():
    print(f"{row.pipeline}, score: {row.score:.4f}")
```